aip/0194.md

```diff
@@ -18,7 +18,8 @@ appropriately.
 ## Guidance
 
 Clients **should** retry requests for which repeated runs would not cause
```

Comment:
What is the distinction between "client", "client library" and "application"?
I could not find these in the glossary and it seems there could be different interpretations.

Here's my interpretation, please let me know what I'm misunderstanding:
In case of the Remote API, Bazel is calling the API using the functions auto generated by java grpc from protos.
If I understand it correctly, the auto generated functions would be the "client library", which should have retry logic but not apply it by default.

Bazel is the "application", which can retry transactional requests (e.g., re-upload all files and try execution again), and has enough context to understand that e.g., INTERNAL errors are fine to retry in the context of RBE.

In this case, what is this "client" entity that should retry UNAVAILABLE but pass through INTERNAL?

Reply:
All of this begs the question: is it useful to document the client library retry semantics in this fashion? I suspect that most API designers and application implementors are more concerned with how application-level retries should work. But maybe that's planned for a separate doc?

Reply:
Seconding! I only understood that you are (apparently) documenting the low-level auto-generated function semantics here from the comment threads (and even now I'm not sure I got that right). Previously, we have referred to this document as guidance for how to design our API and implement the client application logic. The document scope should be clarified.

Reply:
This is a great point, thanks. Will send a clarification and follow up.

Reply:
I have updated the title and language here. For the latter, I mostly changed "retried" to "automatically retried". Does that make the scope clearer?